python/ray/autoscaler/_private/kuberay/run_autoscaler.py

```diff
@@ -5,17 +5,29 @@
 
 import ray
 from ray._private import ray_constants
+from ray._raylet import GcsClient
 from ray._private.ray_logging import setup_component_logger
 from ray._private.services import get_node_ip_address
 from ray._private.utils import try_to_create_directory
 from ray.autoscaler._private.kuberay.autoscaling_config import AutoscalingConfigProducer
 from ray.autoscaler._private.monitor import Monitor
+from ray.autoscaler.v2.instance_manager.config import KubeRayConfigReader
+from ray.autoscaler.v2.utils import is_autoscaler_v2
+from ray.experimental.internal_kv import _initialize_internal_kv
 
 logger = logging.getLogger(__name__)
 
 BACKOFF_S = 5
 
 
+def _get_log_dir() -> str:
+    return os.path.join(
```

Comment:
I don't have time to dive into the get_ray_temp_dir function. Is it hard-coded to /tmp/ray/..., or does it change based on users' configurations?

Reply:
it changes based on user's config, so there's the `ray._private.get_ray_temp_dir` call.